### PR TITLE
Fixing issue #6 and code cleanup.

### PR DESCRIPTION
**Pull request by LinoxGH**
- Removed redundant Integer boxings.
- Fixed the issue #6.
- Removed redundant null check in WorldListener.java.
- Fixed the deprecated handler in Barrels.java.

### CODE REVIEW

File: src/me/john000708/barrels/Barrel.java [hunk 1]
```diff
@@ -42,7 +42,8 @@
     private int capacity;
     private boolean allowDisplayItem;
 
-    public Barrel(Category category, ItemStack item, String name, RecipeType recipeType, final ItemStack[] recipe, int capacity) {
+    // The access can be package-private.
+    Barrel(Category category, ItemStack item, String name, RecipeType recipeType, final ItemStack[] recipe, int capacity) {
```
Comment (TheBusyBiscuit) [inline src/me/john000708/barrels/Barrel.java]: Use protected then. Using the package-private access modifier is not recommended.

Besides, if someone wants to do a subclass for it, they wouldn't have to change it again...

File: src/me/john000708/barrels/Barrels.java [hunk 2]
```diff
@@ -19,13 +19,12 @@
...
 import me.mrCookieSlime.CSCoreLibPlugin.Configuration.Config;
 import me.mrCookieSlime.CSCoreLibPlugin.events.ItemUseEvent;
-import me.mrCookieSlime.CSCoreLibPlugin.general.Inventory.InvUtils;
 import me.mrCookieSlime.CSCoreLibPlugin.general.Inventory.Item.CustomItem;
 import me.mrCookieSlime.Slimefun.Lists.RecipeType;
 import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
 import me.mrCookieSlime.Slimefun.Objects.Category;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
-import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.handlers.ItemInteractionHandler;
+import me.mrCookieSlime.Slimefun.Objects.handlers.ItemInteractionHandler;
```
Comment (TheBusyBiscuit) [inline src/me/john000708/barrels/Barrels.java]: Use the deprecated import for backwards compatibility reasons...

File: src/me/john000708/barrels/Barrels.java [hunk 3]
```diff
@@ -149,8 +152,15 @@ public boolean onRightClick(ItemUseEvent itemUseEvent, Player player, ItemStack
                     Block clickedBlock = itemUseEvent.getClickedBlock();
                     if (BlockStorage.getLocationInfo(clickedBlock.getLocation(), "explosion") == null) {
                         BlockStorage.addBlockInfo(clickedBlock, "explosion", "true");
-                        player.getInventory().setItem(player.getInventory().getHeldItemSlot(), InvUtils.decreaseItem(itemStack, 1));
+                        // Fixes issue #6.
+                        //player.getInventory().setItem(player.getInventory().getHeldItemSlot(), InvUtils.decreaseItem(itemStack, 1));
+                        int amount = itemStack.getAmount();
+                        if (amount <= 1) {
+                            itemStack.setAmount(0);
```
Comment (TheBusyBiscuit) [inline src/me/john000708/barrels/Barrels.java:159]: Did you test this approach?

File: src/me/john000708/barrels/Barrels.java [hunk 4]
```diff
@@ -149,8 +152,15 @@ public boolean onRightClick(ItemUseEvent itemUseEvent, Player player, ItemStack
...
+                        }
+                        else itemStack.setAmount(amount - 1);
                         player.sendMessage(ChatColor.GREEN + "Module successfully applied!");
+                        return true;
```
Comment (TheBusyBiscuit) [inline src/me/john000708/barrels/Barrels.java]: Why was there no return inbefore...????

File: src/me/john000708/barrels/Barrels.java [hunk 5]
```diff
@@ -232,27 +273,35 @@ public boolean onRightClick(ItemUseEvent itemUseEvent, Player player, ItemStack
...
+                List<String> lore = itemStack.getItemMeta().getLore();
 
                 if (lore.get(0).equals("")) {
                     lore.set(0, ChatColor.translateAlternateColorCodes('&', "&0" + player.getUniqueId().toString()));
                     lore.set(1, ChatColor.translateAlternateColorCodes('&', "&fBound to: " + player.getName()));
                     meta.setLore(lore);
-                    idCard.setItemMeta(meta);
+                    itemStack.setItemMeta(meta);
                     player.sendMessage(ChatColor.GREEN + "ID Card bound.");
+                    return true;
```
Comment (TheBusyBiscuit) [inline src/me/john000708/barrels/Barrels.java]: This return statement is redundant, since you return true in the end anyway.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/me/john000708/barrels/Barrels.java (hunk 3 above)
Comment (LinoxGH) [reply src/me/john000708/barrels/Barrels.java:159]: Yes

File: src/me/john000708/barrels/Barrels.java (hunk 5 above)
Comment (LinoxGH) [reply src/me/john000708/barrels/Barrels.java]: But it's return false and we surely want to return true after those happen.

File: src/me/john000708/barrels/Barrels.java (hunk 4 above)
Comment (LinoxGH) [reply src/me/john000708/barrels/Barrels.java]: inbefore?

Comment (TheBusyBiscuit) [reply src/me/john000708/barrels/Barrels.java]: Before your commit, this return true; statement should have been there all along.

File: src/me/john000708/barrels/Barrels.java (hunk 5 above)
Comment (TheBusyBiscuit) [reply src/me/john000708/barrels/Barrels.java]: See Line 304...
You added another return true; statement.

**Comment by LinoxGH** [comment]
Did the requested changes.

**Comment by TheBusyBiscuit** [comment]
Not all.

**Comment by LinoxGH** [comment]
Now all.

**Comment by LinoxGH** [comment]
I am not sure if they are related to my changes but issues #8 and #9 seem like fixed too.

I tried to replicate them in my build but I couldn't.

File: src/me/john000708/barrels/Barrel.java [hunk 6]
```diff
@@ -42,7 +42,8 @@
     private int capacity;
     private boolean allowDisplayItem;
 
-    public Barrel(Category category, ItemStack item, String name, RecipeType recipeType, final ItemStack[] recipe, int capacity) {
+    // The access can be package-private.
```
Comment (TheBusyBiscuit) [inline src/me/john000708/barrels/Barrel.java]: You should maybe update or remove the comment too.

File: src/me/john000708/barrels/Barrels.java (hunk 5 above)
Comment (TheBusyBiscuit) [reply src/me/john000708/barrels/Barrels.java]: This btw is still redundant.

**Comment by LinoxGH** [comment]
Removed the comment and tried to replicate the issue #7 too but it seems that it has been fixed already before in Barrel.java(373:9)

**Comment by TheBusyBiscuit** [comment]
Okay, could you please remove the redundant return-statement too?

**Comment by LinoxGH** [comment]
Is it redundant though? I mean if we don't return true in there it'll return false afterwards.

**Comment by TheBusyBiscuit** [comment]
Yeah but as I said to you before.
You added it twice... You have two return true; statements after another...

File: src/me/john000708/barrels/Barrels.java [hunk 7]
```diff
@@ -232,29 +273,35 @@ public boolean onRightClick(ItemUseEvent itemUseEvent, Player player, ItemStack
...
+                        else itemStack.setAmount(amount - 1);
                         player.sendMessage(ChatColor.GREEN + "Player successfully whitelisted!");
                     } 
                     else {
                         player.sendMessage(ChatColor.RED + "The player is already whitelisted.");
                     }
-
-                }
-                return false;
+                } else return false;
```
Comment (TheBusyBiscuit) [inline src/me/john000708/barrels/Barrels.java]: Don't return false in that case.
Just remove this.

The method has a return false; statement at the very top, If the statement at the top isn't triggered, then this method should always return true.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

**Comment by TheBusyBiscuit** [comment]
Oh now I see what you meant with the return false;
Sorry, I was looking at the wrong return-statement;

We still only need one return true; statement at the very end though.

**Comment by LinoxGH** [comment]
So we return true whatever happens?
Even if the block isn't a barrel owned by the player and the item isn't an ID Card?

**Comment by TheBusyBiscuit** [comment]
No...
The statement at the very top already checks that the item was an ID card.
Otherwise it returns false...

So we can return true; afterwards, no matter what.

Code

**Comment by LinoxGH** [comment]
Ah it's my fault. I always thought the boolean parts of handlers was like the boolean part of CommandExecutor.(You need to return true if you did sth successfully.)

**Comment by TheBusyBiscuit** [comment]
Nah, it's a messy system.

True = This Action was handled
False = This Action is not applicable

On right click it loops through all Handlers until one return true.

**Comment by LinoxGH** [comment]
Return statements should be better now.

**Review by TheBusyBiscuit — APPROVED** [review]

**Pull request merged**